File: src/main/java/org/shark/example/exception/ExceptionAdvice.java
package org.shark.example.exception;

import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.base.pojo.ResponseDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
@Slf4j
public class ExceptionAdvice {

    @ExceptionHandler({Exception.class})
    public final ResponseDto<Void> errorMessageHandler(Exception e) {
        log.error(e.getMessage(), e);
        return ResponseDto.<Void>builder().status(false).errorMessage(e.getMessage()).build();
    }

    @ExceptionHandler({TooManyRequestException.class})
    public final ResponseEntity<String> tooManyRequestErrorMessageHandler(TooManyRequestException e) {
        log.error(e.getMessage(), e);
        return ResponseEntity.status(429).body(e.getMessage());
    }
}

File: src/main/kotlin/org/shark/example/exception/ExceptionAdvice.kt
package org.shark.example.exception

import org.shark.example.service.base.pojo.ResponseDto
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class ExceptionAdvice {

    private val log = LoggerFactory.getLogger(ExceptionAdvice::class.java)

    @ExceptionHandler(Exception::class)
    fun errorMessageHandler(e: Exception): ResponseDto<Void> {
        log.error(e.message, e)
        return ResponseDto(status = false, errorMessage = e.message)
    }
}
